### PR TITLE
MONGOID-4842 use drivers query cache implementation

### PR DESCRIPTION
**Pull request by gjchong25**

### CODE REVIEW

File: lib/mongoid/query_cache.rb [hunk 1]
```diff
@@ -18,7 +18,11 @@ class << self
       #
       # @since 4.0.0
       def cache_table
-        Thread.current["[mongoid]:query_cache"] ||= {}
+        if defined?(Mongo::QueryCache)
```
Comment (johnnyshields) [inline lib/mongoid/query_cache.rb:21]: Mongo is a hard dependency for Mongoid. Why do we need an if-statement to if it's defined? We should always use Mongo::QueryCache

Comment (gjchong25) [reply lib/mongoid/query_cache.rb:21]: Mongo's QueryCache was only recently implemented and available on certain versions, so this check would make sure Mongoid can fall back on its own implementation of QueryCache if the driver's doesn't exist.

Comment (johnnyshields) [reply lib/mongoid/query_cache.rb:21]: How about we just bump the minimum dependency version for `mongo` gem then keep the Mongoid code clean? It should not be a problem for 99.9% of users.

Comment (johnnyshields) [reply lib/mongoid/query_cache.rb:21]: Ah I see you have a discussion with Emily in the JIRA ticket. It's ok. Not how I would do it but ok.

Comment (johnnyshields) [reply lib/mongoid/query_cache.rb:21]: Should raise a ticket as a reminder to remove this code in the Mongoid minor version (7.2) and bump the `mongo` gem dependency at that time.

**Pull request merged**